pymess/backend/dialer/daktela.py

```diff
@@ -0,0 +1,111 @@
+from chamber.utils.datastructures import ChoicesEnum
+from django.utils import timezone
+from django.utils.encoding import force_text
+from django.utils.translation import ugettext_lazy as _l
+
+from pymess.backend.dialer import DialerBackend
+from pymess.config import settings
+from pymess.models import DialerMessage
+from pymess.utils.logged_requests import generate_session
+
+
+class DaktelaDialerBackend(DialerBackend):
+    """
+    Dialer backend implementing Daktela service https://www.daktela.com/api/v6/models/campaignsrecords
+    """
+
+    STATE = ChoicesEnum(
+        ('NOT_ASSIGNED', _l('not assigned'), 'not assigned'),
+        ('READY', _l('ready'), 'ready'),
+        ('RESCHEDULED_BY_DIALER', _l('rescheduled by dialer'), 'rescheduled by dialer'),
+        ('CALL_IN_PROGRESS', _l('call in progress'), 'call in progress'),
+        ('HANGUP', _l('hangup'), 'hangup'),
+        ('DONE', _l('done'), 'done'),
+        ('RESCHEDULED', _l('rescheduled'), 'rescheduled'),
+    )
+
+    # TODO: clean-up mappings
+    STATES_MAPPING = {
+        STATE.NOT_ASSIGNED: DialerMessage.STATE.UNKNOWN,
+        STATE.READY: DialerMessage.STATE.READY,
+        STATE.RESCHEDULED_BY_DIALER: DialerMessage.STATE.RESCHEDULED,
+        STATE.CALL_IN_PROGRESS: DialerMessage.STATE.IN_PROGRESS,
+        STATE.HANGUP: DialerMessage.STATE.DONE,
+        STATE.DONE: DialerMessage.STATE.DONE,
+        STATE.RESCHEDULED: DialerMessage.STATE.RESCHEDULED,
+    }
+
+    SESSION_SLUG = 'pymess-daktela_autodialer'
+
+    @staticmethod
+    def _get_dialer_api_url(name=None):
+        base_url = '/'.join((
+            settings.DIALER_API_URL,
+            settings.DIALER_API_VERSION,
+            settings.DIALER_API_RESOURCE,
+        ))
+        name = '/' + name if name else ''
+
+        return '{base_url}{name}.json?accessToken={access_token}'.format(
+            base_url=base_url, name=name, access_token=settings.DIALER_API_ACCESS_TOKEN,
+        )
+
+    def _update_dialer_states(self, messages):
+        """
+        Method uses Daktela API to get info about autodialer call status
+        :param messages: list of dialer messages to update
+        """
+        for message in messages:
+            client_url = self._get_dialer_api_url(message.tag)
+            result = generate_session(slug=self.SESSION_SLUG, related_objects=(message,)).get(client_url)
+
+            extra_data = message.extra_data.update({
+                'daktela_status': result.status,
+            })
+
+            try:
+                # TODO: check result params and adjust this method if necessary
+                self.update_message(
+                    message,
+                    state=self.STATES_MAPPING[result.status],
+                    error=result.error_message if result.error_message else None,
+                    extra_data=extra_data,
+                )
+            except Exception as ex:
```

Comment:
Neměl bys odchytávat obecnou výjimku.

Reply:
opat je to kopia niecoho existujuceho, co bolo doteraz ok :) neviem, aku konkretnu vynimku tu mam odchytavat. skor by som tam nejako pridaval aj typ, nech sa ulozi, aby sa to dalo postupne konkretizovat

Reply:
už to co píšeš tě má trknout, že je něco špatně. Nic nikdy nekopíruj.

Reply:
nemoze ma to trknut. netusim, ake vsetky vynimky sa mozu udiat. toto zabezpeci aspon to, ze sa akakolvek vynimka zaloguje. ked sa nejaka zachyti, moze sa potom presne specifikovat.

Reply:
Možná teda aspoň re-raisnout?

Reply:
OK

Reply:
tohle by tedy chtělo opravit také v ostatních backendech, kde je tohle dead end a nereraisuje se ...

Reply:
OK

Reply:
byl v tom úmysl, nechtěli jsme aby kvůli odeslání e-mailu, SMS spadl celý request

Reply:
btw když raisneš tu exception tak se nic neuloží do message

Reply:
odchytávat obecnou výjimku by se fakt nemělo nicméně u těch dvou backendů nebylo zdokumentováno v knihovnách, které využívají, které výjimky se vyhazují proto se tam odchytává obecná. Např. u mandrilu tam to vyhazuje asi tucet exceptions s různým předkem. Obecně je to naimplementováno tak, že se zpráva považuje za vytvořenou při prvním pokusu o odeslání, pokud se to nepovede měl by být v cronu task, který to za chvilku zkusí znovu. Proto se tu odchytává všechno možné a nic se nevyhazuje

Reply:
super, ze najprv som to urobit mal, potom zase nie... :) vratim to teda spat

Reply:
@matllubos Potom v kódu mohl být tento komentář aby bylo jasné proč se to děje...